### PR TITLE
Add unit test coverage for PendingAuthListener.

### PR DESCRIPTION
**Pull request by mateusazis**
This class was supposedly already tested via `ServerSecurityPolicyTest.java`, thought that class is actually triggering the short-circuit codepaths (thus, test coverage is reported at 0%).

After this PR, I shall send another one to adapt those tests to force the non-short-circuit async path.

Part of #10566.

### CODE REVIEW

File: binder/src/test/java/io/grpc/binder/internal/PendingAuthListenerTest.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+  @Rule public final MockitoRule mocks = MockitoJUnit.rule();
+
+  @Mock ServerCallHandler<Object, Object> next;
+  @Mock ServerCall<Object, Object> call;
+  @Mock ServerCall.Listener<Object> delegate;
+  @Captor ArgumentCaptor<Status> statusCaptor;
+
+  private final Metadata headers = new Metadata();
+  private final PendingAuthListener<Object, Object> listener = new PendingAuthListener<>();;
```
Comment (jdcormie) [inline binder/src/test/java/io/grpc/binder/internal/PendingAuthListenerTest.java:39]: Extra semicolon. Can you run this through g4 fix or similar?

Comment (mateusazis) [reply binder/src/test/java/io/grpc/binder/internal/PendingAuthListenerTest.java]: Update. Also checked Android Studio, no further warnings open.

File: binder/src/test/java/io/grpc/binder/internal/PendingAuthListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import io.grpc.Metadata;
+import io.grpc.ServerCall;
+import io.grpc.ServerCallHandler;
+import io.grpc.Status;
+
+@RunWith(JUnit4.class)
+
```
Comment (jdcormie) [inline binder/src/test/java/io/grpc/binder/internal/PendingAuthListenerTest.java:28]: Extra line break

**Review by jdcormie — APPROVED** [review]

File: binder/src/test/java/io/grpc/binder/internal/PendingAuthListenerTest.java (hunk 2 above)
Comment (mateusazis) [reply binder/src/test/java/io/grpc/binder/internal/PendingAuthListenerTest.java]: Removed.

**Review by markb74 — APPROVED** [review]

**Pull request merged**